akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -41,19 +41,26 @@ object PersistentActor {
    * Factories for effects - how a persitent actor reacts on a command
    */
   object Effect {
-    def persist[Event, State](event: Event): Effect[Event, State] =
-      new Persist[Event, State](event)
 
-    def persistAll[Event, State](events: im.Seq[Event]): Effect[Event, State] =
+    def persist[Event, State](event: Event, events: Event*): Effect[Event, State] =
+      if (events.nonEmpty)
+        Effect.persist(im.Seq(event) ++ events)
+      else
+        new Persist[Event, State](event)
```

Comment:
I'm keeping `Persist` for the moment because I wanted to check with you guys what you believe is the best. 

Basically, the rationale is to avoid the usage of `PersistentActor.persistAll` down the line. If we know that there is only one event to persist, then we can call `PersistentActor.persist` instead. 

The alterantive is to have an Effect that always have a list of Events and run this kind of optimization later when interacting with the `PersistentActor`.

Reply:
After looking into how `Persist` is implemented, I believe we should remove it and always use `PersistAll`.

Current impl is: 

```scala
private[akka] case class Persist[Event, State](event: Event) extends Effect[Event, State] {
    override val events = event :: Nil
  }
```

But `events` field is never used. When pattern matching in `PersistentActorImpl` we only read the single event passed to the constructor. 

```scala
    case Persist(event) ⇒
      // apply the event before persist so that validation exception is handled before persisting
      // the invalid event, in case such validation is implemented in the event handler.
      state = applyEvent(state, event)
      persist(event) { _ ⇒
        sideEffects.foreach(applySideEffect)
      }
    case PersistAll(events) ⇒
      // apply the event before persist so that validation exception is handled before persisting
      // the invalid event, in case such validation is implemented in the event handler.
      state = events.foldLeft(state)(applyEvent)
      persistAll(scala.collection.immutable.Seq(events)) { _ ⇒
        sideEffects.foreach(applySideEffect)
      }
```

Reply:
As this is internal I think it is fine to keep the little optimization, that `val events` should probably be a `def` or `lazy val` then to not create the list unless it is actually needed.

Reply:
Ok, now I see this one. Ignore my comment below